concepts/observability-strategy/opentelemetry-operator/README.md

```diff
@@ -0,0 +1,89 @@
+
+However, the current functionality of the OpenTelemetry Operator in regards to deployments is very limited. It mainly maps the collector definition 1:1 to the underlying k8s resource. Actual logic which will simplify more complex scenarios are not covered, also simple scenarios like adding a batch processor in combination with a memory-ballast extension with settings fitting to the actual resource settings are not covered. So there is simplification in pure resource handling mainly.
+
+### Guarantees for custom instances
+
+As the OTEL operator will be a feature for users of the cluster, also the lifecycle of users collectors will be managed in regards to image updates (otherwise there is not much meaning in having the feature). Here, the question on the supported image arises. Is it just a plain base image with basic receivers/exporters or the full contrib image? If it ships the full contrib image, there cannot be given any guarantees an updates, it will be best effort only relying on the upstream project fully, it might break your collector. Also providing a hardened image from security perspective will be very challenging as there will be many 3party-dependencies included having no knowledge about actual usage.
+With a stripped-down image version basic tests could be performed to assure that standard scenarios are working and also a security triaging is realistic. However, there will be still not the full guarantee of that an update of your collector instance will be working reliably.
+Providing the base image only brings no value as you rarely can use it without any additional processors. So users would need to opt-out always (which is possible without problems) but again the value of the whole feature is under question.
+
+### The power of auto-instrumentation
+
+Auto-instrumentation sounds like a great idea at a first view. It will provide so much insights of your application without any investment. However, that approach has a high chance of also breaking your application. Especially for nodeJS, the auto feature will just add a node package to an application which can cause version conflicts and block startups.
+Instead, users should add instrumentation (even automatic ones) before the app gets deployed, so that this happens in a controlled and tested way. Again, think of updates to the auto-instrumentation library. If that happens at runtimes, suddenly an application might not startup anymore.
+
+### Injection of settings
+
+The injection of the typical and recommended setting for a workload instrumented by the OTEL SDK is very useful and lowers the burden for managing an application. Insteaf of injecting a hardcoded URL which might change over time, the operator will take care by annotation to inject the proper URL.
```

Comment:
```suggestion
The injection of the typical and recommended setting for a workload instrumented by the OTEL SDK is very useful and lowers the burden for managing an application. Instead of injecting a hardcoded URL (which might change over time), the operator will take care by annotation to inject the proper URL.
```